diff --git a//field_group.module b/field_group.module
index fbf3668..5b0a909 100644
--- a/field_group.module
+++ b/field_group.module
@@ -1866,7 +1866,7 @@ function field_group_build_entity_groups(&$vars, $type) {
   }
 
   // No groups on the entity.
-  if (empty($element['#groups'])) {
+  if (empty($element['#fieldgroups'])) {
     return $element;
   }
 
@@ -1908,7 +1908,7 @@ function field_group_fields_nest(&$element, &$vars = NULL) {
 
   // Create all groups and keep a flat list of references to these groups.
   $group_references = array();
-  foreach ($element['#groups'] as $group_name => $group) {
+  foreach ($element['#fieldgroups'] as $group_name => $group) {
     // Construct own weight, as some fields (for example preprocess fields) don't have weight set.
     $element[$group_name] = array('#child_weight' => -20);
     $group_references[$group_name] = &$element[$group_name];
@@ -1970,7 +1970,7 @@ function field_group_fields_nest(&$element, &$vars = NULL) {
 
   // Bring extra element wrappers to achieve a grouping of fields.
   // This will mainly be prefix and suffix altering.
-  foreach ($element['#groups'] as $group_name => $group) {
+  foreach ($element['#fieldgroups'] as $group_name => $group) {
     field_group_pre_render($group_references[$group_name], $group, $element);
   }
 